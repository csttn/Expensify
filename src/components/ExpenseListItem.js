import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "./numeral";

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <Link className="list-item" to={`/edit/${id}`}>
    <div>
      <h2 className="list-item__title">{description}</h2>
      <span className="list-item__sub-title">
        {moment(createdAt)
          .locale("pt-br")
          .format("Do MMMM, YYYY")}
      </span>
    </div>
    <h3 className="list-item__data">
      {numeral(amount / 100).format("$  0,0.00")}
    </h3>
  </Link>
);

export default ExpenseListItem;
