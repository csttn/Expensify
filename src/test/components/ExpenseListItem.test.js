import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";

test("Renderizando ExpenseLisItem", () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);

  expect(wrapper).toMatchSnapshot();
});
