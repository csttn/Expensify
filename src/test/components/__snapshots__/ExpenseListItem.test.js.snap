// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderizando ExpenseLisItem 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/0"
  >
    <h3>
      Cleyton
    </h3>
  </Link>
  <p>
    R$  1,95
    -
    4th January, 1970
  </p>
</div>
`;
